calibration_assistant.cpp

```diff
@@ -0,0 +1,1250 @@
+                }
+                else
+                {
+                    int recDistance = CalstepDialog::GetCalibrationDistance(pFrame->GetFocalLength(), pCamera->GetCameraPixelSize(), pCamera->Binning);
+                    int currStepSize = TheScope()->GetCalibrationDuration();
+                    int recStepSize;
+                    CalstepDialog::GetCalibrationStepSize(pFrame->GetFocalLength(), pCamera->GetCameraPixelSize(), pCamera->Binning, sidRate,
+                        CalstepDialog::DEFAULT_STEPS, m_currentDec, recDistance, 0, &recStepSize);
+                    if (fabs(1.0 - (double)currStepSize / (double)recStepSize) > 0.3)
+                    {
+                        msg = _("Your current calibration parameters can be adjusted for more accurate results. \n"
+                            "Click on the 'Recal' button to restore them to the default values.");
+                    }
+                }
+                if (msg != wxEmptyString)
+                {
+                    sanityDlg = new CalAssistSanityDialog(this, msg);
+                    sanityDlg->ShowModal();
+                }
+            }
+        }
+    }
+}
+
+void CalibrationAssistant::ShowError(const wxString& msg, bool fatal)
+{
+    m_pMessage->SetLabelText(msg);
+    if (fatal)
+    {
+        m_pSlewBtn->Enable(false);
+        m_pCalibrateBtn->Enable(false);
+
+    }
+}
+
+void CalibrationAssistant::ShowStatus(const wxString& msg)
+{
+    m_pMessage->SetLabelText(msg);
+}
+
+double CalibrationAssistant::GetCalibrationDec()
+{
+    return m_currentDec;
+}
+
+void CalibrationAssistant::TrackCalibration(GUIDER_STATE state)
+{
+    switch (state)
+    {
+    case STATE_UNINITIALIZED:
+    case STATE_SELECTING:
+    case STATE_SELECTED:
+        if (m_calibrationActive)
+        {
+            ShowStatus(_("Calibration failed or was cancelled"));
+            m_lastResult = "Incomplete";
+            m_calibrationActive = false;
+            m_monitoringCalibration = false;
+            m_pSlewBtn->Enable(true);
+            m_pCalibrateBtn->Enable(true);
+            m_pExplainBtn->Enable(true);
+        }
+        break;
+    case STATE_CALIBRATING_PRIMARY:
+    case STATE_CALIBRATING_SECONDARY:
+        m_calibrationActive = true;
+        break;
+    case STATE_CALIBRATED:
+    case STATE_GUIDING:
+    {
+        m_calibrationActive = false;
+        m_monitoringCalibration = false;
+        EvaluateCalibration();
+        m_pCalibrateBtn->Enable(true);
+        break;
+    }
+
+    }
+}
+
+void CalibrationAssistant::OnTimer(wxTimerEvent& evt)
+{
+    if (!m_monitoringCalibration)
+        UpdateCurrentPosition(true);
+    else
+        TrackCalibration(pFrame->pGuider->GetState());
+}
+
+void CalibrationAssistant::UpdateCurrentPosition(bool fromTimer)
+{
+    double ra, dec, lst;
+    double hourAngle;
+    const int INVALID_DEC = 100;
+    static double lastDec = INVALID_DEC;
+
+    if (!pPointingSource->GetCoordinates(&ra, &dec, &lst))

```

Comment:
I'd also decrease indentations here in the same way (I find deep indentation difficult to read since I lose the context of the branches)
```
    bool error = pPointingSource->GetCoordinates(&ra, &dec, &lst);
    if (error)
    {
        // show error message
        return;
    }
    // ... ordinary non-error code follows ...
    ...
    if (m_meridianFlipping)
        return;
    ... ordinary non-meridian flipping stuff ...
    etc
}